linux/EditorWidget-linux.cpp

```diff
@@ -0,0 +1,53 @@
+/*****************************************************************************
+Copyright (C) 2016-2017 by Colin Edwards.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*****************************************************************************/
+
+#include "../headers/EditorWidget.h"
+
+#include <QWindow>
+#include <X11/Xlib.h>
+
+void EditorWidget::buildEffectContainer(AEffect *effect) {
+	WNDCLASSEX wcex{ sizeof(wcex) };
+	wcex.lpfnWndProc = DefWindowProc;
+	wcex.hInstance = GetModuleHandle(0);
+	wcex.lpszClassName = L"Minimal VST host - Guest VST Window Frame";
+	RegisterClassEx(&wcex);
+
+	const auto style = WS_CAPTION | WS_THICKFRAME | WS_OVERLAPPEDWINDOW;
+	HWND hwnd = CreateWindow(
+		wcex.lpszClassName, TEXT(""), style
+		, 0, 0, 0, 0, 0, 0, 0, 0
+	);
+
+	QWidget *widget = QWidget::createWindowContainer(QWindow::fromWinId((WId)hwnd), this);
+	widget->move(0, 0);
+	widget->resize(300, 300);
+
+	effect->dispatcher(effect, effEditOpen, 0, 0, hwnd, 0);
+
+	VstRect* vstRect = nullptr;
+	effect->dispatcher(effect, effEditGetRect, 0, 0, &vstRect, 0);
+	if (vstRect)
+	{
+		widget->resize(vstRect->right - vstRect->left, vstRect->bottom - vstRect->top);
+	}
+}
+
+void EditorWidget::handleResizeRequest(int width, int height) {
+	// We don't have to do anything here as far as I can tell. The widget will resize the HWIND itself and then
```

Comment:
I'm trying to find out. The linux stuff isn't complete by a long shot, but it's 'better' in that it compiles and runs, it just crashes when one actually selects a vst plugin.